packages/terra-data-grid/src/subcomponents/ColumnResizeHandle.jsx

```diff
@@ -85,28 +114,115 @@ const ColumnResizeHandle = (props) => {
+    const parentTable = resizeHandleRef.current.closest('table');
+
+    // Update resize handle height to match parent table height
+    if (parentTable) {
+      resizeHandleRef.current.style.height = `${parentTable.offsetHeight}px`;
+    }
+
+    // Assistive technologies should announce aria-label text once focused
+    setIsAriaLabel(true);
+  };
+
+  const onMouseLeave = () => {
+    if (document.activeElement !== resizeHandleRef.current) {
+      resizeHandleRef.current.style.height = '100%';
+    }
+  };
+
+  // Handle column resize handle keyboard navigation
+  const onKeyDown = (event) => {
+    const key = event.keyCode;
+    switch (key) {
+      case KeyCode.KEY_SPACE:
+      case KeyCode.KEY_RETURN:
+        // Lock focus into component
+        resizeHandleRef.current.focus();
+        setNavigationEnabled(false);
+        // Assistive technologies should avoid announcing aria-label while focus locked, but announce aria-valueText instead
+        setIsAriaLabel(false);
+        setIsAriaValueText(true);
+        // Assistive technologies should make announcement for focus locked
+        columnContext.setColumnHeaderAriaLiveMessage(intl.formatMessage({ id: 'Terra.worklist-data-grid.cell-focus-trapped' }));
+        event.stopPropagation();
+        event.preventDefault();
+        break;
+      case KeyCode.KEY_ESCAPE:
+        // Assistive technologies should stop announcing aria-valueText once focus unlocked
+        setIsAriaValueText(false);
+        // Release focus lock
+        columnContext.setColumnHeaderAriaLiveMessage(intl.formatMessage({ id: 'Terra.worklist-data-grid.resume-navigation' }));
+        setNavigationEnabled(true);
+        break;
+      case KeyCode.KEY_RIGHT:
+        if (onResizeHandleChange && !isNavigationEnabled) {
+          // Increase column width
+          onResizeHandleChange(columnIndex, columnResizeIncrement);
+        }
+        break;
+      case KeyCode.KEY_LEFT:
+        if (onResizeHandleChange && !isNavigationEnabled) {
+          // Decrease column width
+          onResizeHandleChange(columnIndex, -columnResizeIncrement);
+        }
+        break;
+      default:
+    }
+
+    if (!isNavigationEnabled) {
+      event.stopPropagation();
+      event.preventDefault();
+    }
+  };
+
+  // Prevent click event propagation
+  const onClick = (event) => {
+    // Prevent event bubbling since necessary actions are handled by this component
+    event.stopPropagation();
+  };
+
+  const onBlur = () => {
+    setNavigationEnabled(true);
+    setIsAriaLabel(false);
+  };
+
   return (
-    /* eslint-disable react/forbid-dom-props */
-    // eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/interactive-supports-focus
+    // eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions, jsx-a11y/no-static-element-interactions
     <div
       ref={resizeHandleRef}
       draggable
-      role="slider"
+      role={!isNavigationEnabled ? 'slider' : 'divider'}
```

Comment:
Should we use isNavigationEnabled instead of "!isNavigationEnabled" to deal with the positive case in the comparison?

Reply:
reversed condition here: 04a321ed7dfea78806cae4df33e09d43c17bb022